lib/qbwc/session.rb

```diff
@@ -56,12 +67,10 @@ def response=(qbxml_response)
       QBWC.logger.info 'Parsing response.'
       response = QBWC.parser.from_qbxml(qbxml_response)["qbxml"]["qbxml_msgs_rs"].except("xml_attributes")
       response = response[response.keys.first]
-      QBWC.logger.info 'Parsing headers.'
       parse_response_header(response)
-      is_error = (self.error && self.status_severity == 'Error')
-      QBWC.logger.info "Processing response."
-      self.current_job.process_response(response, self, iterator_id.blank? && (!is_error || QBWC::on_error == 'continueOnError'))
-      self.next_request unless is_error || self.iterator_id.present? # search next request
+      self.current_job.process_response(response, self, iterator_id.blank?) unless self.current_job.nil? || error_and_stop_requested?
```

Comment:
Yes I think you're right; there's no reason the app shouldn't receive control; this might even be an opportunity to automatically recover and/or prevent a stuck-on-error scenario.